### PR TITLE
added timezone parameter to dbconnect method

### PR DESCRIPTION
**Pull request by ycphs**
allow to use time zones other than UTC.

My suggestion would be the system time zone as default. for backward compatibility "+00:00" could be used as well.

### CODE REVIEW

File: R/connect.R [hunk 1]
```diff
@@ -99,7 +103,18 @@ setMethod("dbConnect", "MariaDBDriver",
       bigint = bigint
     )
 
-    dbExecute(conn, "SET time_zone = '+00:00'")
+    if(exists('timezone')) {
```
Comment (krlmlr) [inline R/connect.R:106]: Can you please add a `timezone = "+0000"` argument to the function's formals (inside `function()`)? Then you can test with

```suggestion
    if (!missing("timezone")) {
```

and also avoid one `dbExecute()` call below.

**Review by krlmlr — COMMENTED** [review]
Thanks for working on this!

**Comment by krlmlr** [comment]
Can you also please merge/rebase with master?

File: R/connect.R (hunk 1 above)
Comment (ycphs) [reply R/connect.R]: I removed the `dbExecute(conn, "SET autocommit = 0")`, but are you sure that it shouldn't be there?

File: R/connect.R [hunk 2]
```diff
@@ -99,8 +103,19 @@ setMethod("dbConnect", "MariaDBDriver",
...
+        ,
+        error = function(e) {
+          dbExecute(conn, "SET time_zone = '+00:00'")
+          message("Unknown or incorrect time zone: '", timezone, "'")
+        }
+      )
+    } else {
+      dbExecute(conn, "SET time_zone = '+00:00'")
+    }
+
```
Comment (krlmlr) [inline R/connect.R:118]: I misread -- failure to set the timezone really should be an error.

```suggestion
    dbExecute(conn, paste0("SET time_zone = ", dbQuoteString(conn, timezone))
    dbExecute(conn, "SET autocommit = 0")
```

File: R/connect.R [hunk 3]
```diff
@@ -99,8 +103,19 @@ setMethod("dbConnect", "MariaDBDriver",
...
     conn
```
Comment (krlmlr) [inline R/connect.R:119]:
```suggestion
    on.exit(NULL)

    conn
```

**Comment by krlmlr** [comment]
What's a good test for the new argument?

File: R/connect.R [hunk 4]
```diff
@@ -99,8 +103,10 @@ setMethod("dbConnect", "MariaDBDriver",
       bigint = bigint
     )
```
Comment (krlmlr) [inline R/connect.R:105]:
```suggestion

    on.exit(dbDisconnect(conn))

```

File: man/dbConnect-MariaDBDriver-method.Rd [hunk 5]
```diff
@@ -23,7 +23,8 @@
   ssl.cipher = NULL,
   ...,
   bigint = c("integer64", "integer", "numeric", "character"),
-  timeout = 10
+  timeout = 10,
+  timezone = "+0000"
```
Comment (krlmlr) [inline man/dbConnect-MariaDBDriver-method.Rd:27]:
```suggestion
  timezone = "+00:00"
```

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**